ideascube/tests/test_import_users.py

```diff
@@ -0,0 +1,222 @@
+    response = form.submit()
+    response.follow()
+    assert User.objects.count() == 2
+    user = User.objects.order_by('pk').last()
+    assert user.serial == '12345'
+    assert user.camp_entry_date.year == 2016
+    assert user.camp_entry_date.month == 10
+    assert user.camp_entry_date.day == 27
+
+
+def test_should_update_existing_users(staffapp, monkeypatch):
+    data = (b'serial,full_name\n'
+            b'12345,Gabriel Garcia Marquez')
+    UserFactory(full_name='Gabriel Garcia Marques', serial='12345')
+    assert User.objects.count() == 2
+    form = staffapp.get(reverse('user_import')).forms['import']
+    form['source'] = Upload('users.csv', data, 'text/csv')
+    response = form.submit()
+    response.follow()
+    assert User.objects.count() == 2
+    user = User.objects.order_by('pk').last()
+    assert user.serial == '12345'
+    assert user.full_name == 'Gabriel Garcia Marquez'  # s has been fixed in z.
+
+
+def test_should_not_replace_is_staff(staffapp, monkeypatch):
+    data = (b'serial,full_name,is_staff\n'
+            b'12345,Gabriel Garcia Marquez,True\n'
+            b'23456,Miguel de Cervantes,False')
+    UserFactory(serial='12345', is_staff=False)
+    UserFactory(serial='23456', is_staff=True)
+    form = staffapp.get(reverse('user_import')).forms['import']
+    form['source'] = Upload('users.csv', data, 'text/csv')
+    response = form.submit()
+    response.follow()
+    user = User.objects.order_by('pk')[1]
+    assert user.serial == '12345'
+    assert user.full_name == 'Gabriel Garcia Marquez'
+    assert user.is_staff is False
+    user = User.objects.order_by('pk')[2]
+    assert user.is_staff is True
+    assert user.full_name == 'Miguel de Cervantes'
+
+
+def test_should_not_set_password(staffapp, monkeypatch):
+    data = (b'serial,full_name,password\n'
+            b'12345,Gabriel Garcia Marquez,123456')
+    form = staffapp.get(reverse('user_import')).forms['import']
+    form['source'] = Upload('users.csv', data, 'text/csv')
+    response = form.submit()
+    response.follow()
+    user = User.objects.order_by('pk')[1]
+    assert user.serial == '12345'
+    assert user.full_name == 'Gabriel Garcia Marquez'
+    assert not user.password
+
+
+def test_should_not_change_password(staffapp, monkeypatch):
+    data = (b'serial,full_name,password\n'
+            b'12345,Gabriel Garcia Marquez,123456')
+    before = UserFactory(serial='12345', is_staff=False)
+    form = staffapp.get(reverse('user_import')).forms['import']
+    form['source'] = Upload('users.csv', data, 'text/csv')
+    response = form.submit()
+    response.follow()
+    user = User.objects.order_by('pk')[1]
+    assert user.serial == '12345'
+    assert user.full_name == 'Gabriel Garcia Marquez'
+    assert user.password == before.password
+
+
+@pytest.mark.usefixtures('systemuser')
+def test_does_not_try_to_create_system_user(staffapp, monkeypatch):
+    qs = User.objects.all(include_system_user=True)
```

Comment:
Aren't we removing the `.all(include_system_user=True)` API in favour of the `.get_unfiltered_queryset()` one?

https://github.com/ideascube/ideascube/pull/616#discussion_r85548778